libcloud/dns/drivers/rcodezero.py

```diff
@@ -0,0 +1,506 @@
+        RecordType.SOA: 'SOA',
+        RecordType.SPF: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.SSHFP: 'SSHFP',
+        RecordType.SRV: 'SRV',
+        RecordType.TLSA: 'TLSA',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, key, secret=None, secure=True, host='my.rcodezero.at',
+                 port=None, api_version='v1', **kwargs):
+        """
+        :param    key: API token to be used (required)
+        :type     key: ``str``
+
+        :param    secure: Whether to use HTTPS (default) or HTTP.
+        :type     secure: ``bool``
+
+        :param    host: Hostname used for connections.
+                        Default: ``my.rcodezero.at``.
+        :type     host: ``str``
+
+        :param    port: Port used for connections.
+        :type     port: ``int``
+
+        :param    api_version: Specifies the API version to use.
+                               ``v1`` is currently the only valid
+                               option (and default)
+        :type     api_version: ``str``
+
+        :return: ``None``
+        """
+
+        if api_version == 'v1':
+            self.api_root = '/api/v1'
+        else:
+            raise NotImplementedError('Unsupported API version: %s' %
+                                      api_version)
+
+        super(RcodeZeroDNSDriver, self).__init__(key=key, secure=secure,
+                                                 host=host, port=port,
+                                                 **kwargs)
+
+    def create_record(self, name, zone, type, data, extra=None):
+        """
+        Create a new record in a given, existing zone.
+
+        :param name: name of the new record without the domain name,
+                     for example "www".
+        :type  name: ``str``
+
+        :param zone: Zone in which the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS resource record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depending on the record type).
+        :type  data: ``str``
+
+        :param extra: Extra attributes: 'ttl', 'disabled'
+        :type   extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        action = '%s/zones/%s/rrsets' % (self.api_root, zone.id)
+
+        payload = self._to_patchrequest(
+            zone.id, None, name, type, data, extra, 'add')
+
+        try:
+            self.connection.request(action=action, data=json.dumps(payload),
+                                    method='PATCH')
+        except BaseHTTPError:
+            e = sys.exc_info()[1]
+            if e.code == httplib.UNPROCESSABLE_ENTITY and \
+               e.message.startswith('Could not find domain'):
+                raise ZoneDoesNotExistError(zone_id=zone.id, driver=self,
+                                            value=e.message)
+            raise e
+
+        if not (extra is None or extra.get('ttl', None) is None):
```

Comment:
Can you refactor this line please, I had to read it 3 times to understand the logic, unless I need another ☕️ :-)

Reply:
fixed in b9f9bf85fda6ad94c3323c24f67829b139784b06